geoopt/manifolds/base.py

```diff
@@ -842,9 +842,27 @@ def _retr_transp(self, x, v, *more, u, t):
                 out = (y, self._transp2y(x, v, *more, y=y))
         return out
 
+    def rand_(self, x):
+        """
+        Naive implemetation for normal distribution
```

Comment:
Should comment that this is an inplace version

Reply:
`self.rand` is still missing, probably it is in `TODO` now. I suggest not to use any inplace operations in non inplace version. That would allow reparametrization on manifold parameters